store/store.go

```diff
@@ -1786,11 +1793,18 @@ func (s *Store) updateAppliedIndex() chan struct{} {
 	return done
 }
 
-func (s *Store) isStaleRead(freshness int64) bool {
-	if freshness == 0 || s.raft.State() == raft.Leader {
+func (s *Store) isStaleRead(freshness, maxStale int64) bool {
+	if s.raft.State() == raft.Leader {
 		return false
 	}
-	return time.Since(s.raft.LastContact()).Nanoseconds() > freshness
+	return IsStaleRead(
```

Comment:
What is the advantage of splitting into two functions?

Reply:
Testability.